### DISCUSSION TITLE
Could we still output `--stats` when the process is interrupted?

### DISCUSSION THREAD

**Discussion by tomwadeson** [BurntSushi/ripgrep#2155] [General] [answered]
If `rg` runs with the `--stats` option and is interrupted before it completes, would it be possible to output the aggregate statistics up until the point that the process was killed/interrupted?

**Comment by BurntSushi** [comment, marked as answer]
AIUI, the way to do this is to install a signal handler. A while back, we had a signal handler installed that would "clear the terminal" to avoid scenarios where ripgrep was killed in the middle of writing colored text to the terminal. When that happens, the coloring styling stays enabled for the terminal even though ripgrep has stopped. So the signal handler would ensure that terminal coloring was "reset." That was done in this PR: https://github.com/BurntSushi/ripgrep/pull/187

However, the way it was implemented seemed to not work in certain situations on Windows, and basically had the effect of making `^C` do _nothing_. That can never be acceptable. Above all else, users need the ability to kill ripgrep reliably if they want it to stop. This was discussed here: https://github.com/BurntSushi/ripgrep/issues/281 And the change to remove `^C` handling was made here: https://github.com/BurntSushi/ripgrep/commit/de5cb7d22e00c18dfce082a68b0cf45924b56e7f

While that discussion tended to focus on coloring, this one does not and is perhaps a little simpler to deal with. However, the problem of doing signal handling correctly and reliably still stands. Basically, if this were to be done it would need to be very thoroughly vetted and tested across Unix and Windows.
